filters/include/pcl/filters/impl/sampling_surface_normal.hpp

```diff
@@ -212,7 +212,7 @@ pcl::SamplingSurfaceNormal<PointT>::computeNormal (const PointCloud& cloud, Eige
 
   if (computeMeanAndCovarianceMatrix (cloud, covariance_matrix, xyz_centroid) == 0)
   {
-    nx = ny = nz = curvature = std::numeric_limits<float>::quiet_NaN ();
+    curvature = std::numeric_limits<float>::quiet_NaN ();
```

Comment:
I think the aim was to modify the `normal` too.

Reply:
https://github.com/PointCloudLibrary/pcl/blob/014926f4ce898d5ccf7a7767a0824f02764c4248/filters/include/pcl/filters/impl/sampling_surface_normal.hpp#L220-L227

`normal (3) =0;` can be removed, anything else?

Reply:
This?

https://github.com/PointCloudLibrary/pcl/blob/014926f4ce898d5ccf7a7767a0824f02764c4248/filters/include/pcl/filters/impl/sampling_surface_normal.hpp#L219-L227
```
                                                   |
                                                  \ /
  // Get the plane normal and surface curvature
  normal (0) = 0.0f; // nx
  normal (1) = 0.0f; // ny
  normal (2) = 0.0f; // nz
  solvePlaneParameters (covariance_matrix, normal (0), normal (1), normal (2), curvature);
  // Hessian form (D = nc . p_plane (centroid here) + p)
  normal (3) = -1 * normal.dot (xyz_centroid);
```

Reply:
I meant make the normal also contain `quiet_NaN` or 0 (Need to see what the original implementer intended)

Reply:
https://github.com/PointCloudLibrary/pcl/blob/014926f4ce898d5ccf7a7767a0824f02764c4248/filters/include/pcl/filters/impl/sampling_surface_normal.hpp#L213-L217

It's returning after setting it to `quiet_NaN`, so the values of `nx`, `ny`, `nz` is not used, so `normal` shouldn't be changed in the if clause?

Reply:
curvature and normal are out variables.

IMO, the behavior should be uniform (and the docs don't say anything regarding the case with `NaN` curvature). If curvature is `NaN`, what should the normal be? `0,0,0,1`? List of `NaN`s? untouched? In some places, PCL leaves out variable untouched. In others like this, I have no clue.

Needs input from @SergioRAgostinho or @taketwo